.github/workflows/e2e-tests-boost.yml

```
@@ -2,29 +2,37 @@ name: Jetpack Boost E2E Tests
 
 on:
   pull_request:
-    paths:
-      - 'projects/plugins/boost/**'
-      - '!projects/plugins/boost/**.md'
-      - '!projects/plugins/boost/**.txt'
-      - '!projects/plugins/boost/docs/**'
-      - '!projects/plugins/boost/changelog/**'
-  push:
-    branches: [master]
-    paths:
-      - 'projects/plugins/boost/**'
-      - '!projects/plugins/boost/**.md'
-      - '!projects/plugins/boost/**.txt'
-      - '!projects/plugins/boost/docs/**'
-      - '!projects/plugins/boost/changelog/**'
+    push:
+      branches: [ master ]
```

Bad indentation, which breaks the YAML. I think the result of this would be that the tests no longer run on push at all.
```suggestion
  push:
    branches: [ master ]
```

---

Thanks for spotting this one. Fixed it in ba4721a

---

Also, I fixed the issue with the "Trigger test report workflow" job despite the test job being skipped in commit ca2a7ec.
Actually, I am not entirely sure why it did run before as the "Slack notification" did not run with only the `needs: e2e-tests` statement which the "Trigger test report workflow" job had too. Anyway it seems that the fix in ca2a7ec does the job.

---

Regarding why the Trigger test report workflow" job despite the test job being skipped, I am not sure to understand why because the "Slack Notification" job is being skipped. Both of them have the `needs: e2e-tests` statement. The only difference was that the position in the definition which I fixed with commit b647b5e and still does not work. I also experimented with the `if` statement for the `test-reports` job in ca2a7ec but it seems to have broken the workflow and even the "Check if Boost E2E tests should run" check got skipped so I reverted it back. So I don't have a solution for this and I am not sure whether or not it is an issue to have the reports job run nevertheless the tests job ran or not.

---

The "Trigger test report workflow" job runs because the e2e-tests job being skipped doesn't count as "workflow cancelled", so the `if: ${{ ! cancelled() }}` check passes.

I think ca2a7ec didn't work because the job only has `needs: e2e-tests`, so `check` isn't part of the `needs` context. You could probably make it work with something like `if: ${{ ! cancelled() && needs.e2e-tests.result != 'skipped' }}`. Or by doing what you did in ca2a7ec while also setting `needs: [ check, e2e-tests ]`, but checking the e2e-tests result seems cleaner to me.

---

Got it now, I am understanding GitHub actions a lot more. Thank you @anomiex for the guidance.

I have made the suggested change using the e2e-tests result check in commit f917f76 and it is working as expected now.